src/v2/Carousel/Carousel.sol

```diff
@@ -394,35 +394,40 @@ contract Carousel is VaultV2 {
         while ((index - prevIndex) < (_operations)) {
             // only roll over if last epoch is resolved
             if (epochResolved[queue[index].epochId]) {
-                uint256 entitledShares = previewWithdraw(
+                uint256 entitledAmount = previewWithdraw(
                     queue[index].epochId,
                     queue[index].assets
                 );
                 // mint only if user won epoch he is rolling over
-                if (entitledShares > queue[index].assets) {
+                if (entitledAmount > queue[index].assets) {
+                uint256 diff = entitledAmount - queue[index].assets;
+                // get diff amount in assets 
+                uint256 diffInAssets = diff.mulDivUp(finalTVL[queue[index].epochId], claimTVL[queue[index].epochId]);
                     // skip the rollover for the user if the assets cannot cover the relayer fee instead of revert.
                     if (queue[index].assets < relayerFee) {
                         index++;
                         continue;
                     }
+
+                    uint256 originalDepositValue = queue[index].assets - diffInAssets;
```

Comment:
Highly profitable epochs could underflow here since diffInAssets can be greater than queue[index].assets. I.e. entitled amount = 200 and assets = 99. diff = 101 and 99 - 101 will underflow. That edge case would break rollover functionality. Mitigation would be complex so maybe better to just acknowledge and accept the risk

Reply:
@IAm0x52 issues does not exist in later PR, see lates PR: https://github.com/Y2K-Finance/Earthquake/blob/5cb2890e9826b06fff95014cf8b016b4f441dd27/src/v2/Carousel/Carousel.sol#L422

Here i convert profit in nominal terms back to shares which is lower than nominal terms of `entitledAmount`